odl/tomo/geometry/spect_geometry.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2014 - 2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""SPECT geometry."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+import numpy as np
+
+# Internal
+import odl
+from odl.tomo.geometry.detector import Flat2dDetector
+from odl.tomo.geometry.geometry import (Geometry,
+                                        AxisOrientedGeometry)
+from odl.tomo.util.utility import perpendicular_vector
+
+__all__ = ('ParallelHoleCollimatorGeometry', )
+
+
+class ParallelHoleCollimatorGeometry(Geometry, AxisOrientedGeometry):
```

Comment:
I'm not quite happy with the name. What is this "hole" about anyway? We should think about the different geometry types that are used in SPECT and what distinguishes them. We don't have to repeat all kinds of technical details.

So what is characteristic of this geometry? Mathematically, it's (correct me if I'm wrong) in the simplest case just a standard single-axis parallel 3D geometry. To take the imperfect collimation into account, one needs the additional information of the detector position and some PSF parameters. In the full-blown and most generic setting, each detector pixel has its own field of view, which means that there is not a unique direction of incoming rays, but a bunch of directions which are integrated over. Our general geometry model would allow that, but I don't think any forward projector supports that.

So my conclusion is that

- this geometry should subclass `Parallel3dAxisGeometry` and by default be identical to it,
- offer *optional* parameters for detector positioning,
- offer *optional* parameters for the collimators.

And we should make an attempt at understanding what the SPECT forward model really is about. Where does this PSF come from and on which parameters does it depend?